src/tools/sfdk/commandlineparser.cpp

```diff
@@ -168,6 +176,18 @@ CommandLineParser::CommandLineParser(const QStringList &arguments)
         return;
     }
 
+    for (const QString &value : parser.values(COption)) {
+        // setCurrentWorkingDirectory() cannot be used before SdkManager is instantiated
+        m_cwdChanges += [=]() -> bool {
+            QString errorString;
+            if (!setCurrentWorkingDirectory(value, &errorString)) {
+                badUsage(invalidArgumentToOptionMessage(errorString, COption.names().first(), value));
+                return false;
+            }
+            return true;
+        };
```

Comment:
Wouldn't it be simpler to store just the directories in m_cwdChanges, as a stringlist? Feels kind of silly to store these functions instead, when it is the same function for each and every directory.

Reply:
It needs to access the COption variable.

Reply:
In order to not just hardcode "C" here? I'm a bit undecided here which one is worse, the disease or the cure. Well, it's your code. I just wish I don't have to fix bugs here.

Reply:
It's just a task executed asynchronously, waiting for a precondition. Compared to spreading the data and code throughout the CommandLineParser class, this way the task is nicely encapsulated in its own class (lambdas are kind of functor classes).

Reply:
Yeah, there is value there. It definitely tips the scale towards your approach. I'm still not sure if it tips it enough that I would choose the approach though ;)